blackjax/tempered_smc.py

```diff
@@ -0,0 +1,116 @@
+"""Public API for the Tempered SMC Kernel"""
+from typing import Callable, Dict, List, NamedTuple, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+
+import blackjax.inference.base as base
+from blackjax.hmc import HMCParameters
+
+Array = Union[np.ndarray, jnp.DeviceArray]
+PyTree = Union[Dict, List, Tuple]
+
+__all__ = ["new_state", "kernel"]
+
+
+class SMCState(NamedTuple):
+    """ Current state for the tempered SMC algorithm. This encodes the current set of sampled particles
+    as well as the corresponding tempering parameter :math:`\lambda`
+    """
+    particles: PyTree[Array]
+    lamda: float
+
+
+class SMCInfo(NamedTuple):
```

Comment:
We should propagate the info that comes from the MCMC algorithm, in a `mcmc_info` field. Can be useful for debugging.

Reply:
np